fix(linter/react): `exhaustive-deps` report longest dependency

overlookmotel opened the pull request:
Where a hook has multiple overlapping dependencies, `react/exhaustive-deps` rule reports that overlap. e.g.:

```js
function Foo(props) {
    useCallback(() => {
        console.log(props.foo);
    }, [props, props.foo]);
    return <div />;
}
```

```
eslint-plugin-react(exhaustive-deps): React Hook useCallback has unnecessary dependency: props.foo
```

[eslint-plugin-react-hooks](https://www.npmjs.com/package/eslint-plugin-react-hooks) always reports the longest dependency (`props.foo` in above case). Whereas Oxlint would report either `props` or `props.foo` randomly - because the dependencies are stored in an `FxHashSet` which does not have a defined iteration order.

This PR fixes that so it always reports the longer dependency, aligning with the ESLint plugin.

overlookmotel commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9891?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#9886** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9886?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#9883** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9883?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#9891** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9891?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9891?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#9885** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9885?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#9884** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9884?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=9891).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Boshen commented:
### Merge activity

* **Mar 19, 9:02 PM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
fix(linter/react): `exhaustive-deps` report longest dependency (#9891)

Where a hook has multiple overlapping dependencies, `react/exhaustive-deps` rule reports that overlap. e.g.:

```js
function Foo(props) {
    useCallback(() => {
        console.log(props.foo);
    }, [props, props.foo]);
    return <div />;
}
```

```
eslint-plugin-react(exhaustive-deps): React Hook useCallback has unnecessary dependency: props.foo
```

[eslint-plugin-react-hooks](https://www.npmjs.com/package/eslint-plugin-react-hooks) always reports the longest dependency (`props.foo` in above case). Whereas Oxlint would report either `props` or `props.foo` randomly - because the dependencies are stored in an `FxHashSet` which does not have a defined iteration order.

This PR fixes that so it always reports the longer dependency, aligning with the ESLint plugin.
